qiskit/circuit/quantumcircuit.py

```diff
@@ -325,39 +325,193 @@ def draw(self, scale=0.7, filename=None, style=None, output='text',
                                             justify=justify)
 
     def size(self):
-        """Return total number of operations in circuit."""
-        # TODO: removed the DAG from this function
-        from qiskit.converters import circuit_to_dag
-        dag = circuit_to_dag(self)
-        return dag.size()
+        """Returns total number of gate operations in circuit.
+
+        Returns:
+            int: Total number of gate operations.
+        """
+        gate_ops = 0
+        for item in self.data:
+            if item.name not in ['barrier', 'snapshot']:
+                gate_ops += 1
+        return gate_ops
 
     def depth(self):
-        """Return circuit depth (i.e. length of critical path)."""
-        from qiskit.converters import circuit_to_dag
-        dag = circuit_to_dag(self)
-        return dag.depth()
+        """Return circuit depth (i.e. length of critical path).
+        This does not include compiler or simulator directives
+        such as 'barrier' or 'snapshot'.
+
+        Returns:
+            int: Depth of circuit.
+
+        Notes:
+            The circuit depth and the DAG depth need not bt the
+            same.
+        """
+        # Labels the registers by ints
+        # and then the qubit position in
+        # a register is given by reg_int+qubit_num
+        reg_offset = 0
+        reg_map = {}
+        for reg in self.qregs+self.cregs:
+            reg_map[reg.name] = reg_offset
+            reg_offset += reg.size
+
+        # A list that holds the height of each qubit
+        # and classical bit.
+        op_stack = [0]*reg_offset
+        # Here we are playing a modified version of
+        # Tetris where we stack gates, but multi-qubit
+        # gates, or measurements have a block for each
+        # qubit or cbit that are connected by a virtual
+        # line so that they all stacked at the same depth.
+        # Conditional gates act on all cbits in the register
+        # they are conditioned on.
+        # We do not consider barriers or snapshots as
+        # They are transpiler and simulator directives.
+        # The max stack height is the circuit depth.
+        for op in self.data:
+            if op.name not in ['barrier', 'snapshot']:
+                levels = []
+                reg_ints = []
+                for ind, reg in enumerate(op.qargs+op.cargs):
```

Comment:
After #1816 the ops no longer have .qargs and .cargs. Instead, when you iterate over `circuit.data` you get tuples of `(op, qargs, cargs)`. So this PR needs to be updated.

Reply:
resolved